crates/rpc/rpc/src/eth/api/block.rs

```diff
@@ -12,6 +12,17 @@ impl<Client, Pool, Network> EthApi<Client, Pool, Network>
 where
     Client: BlockProvider + StateProviderFactory + 'static,
 {
+    /// Returns the uncle headers of the given block
+    ///
+    /// Returns an empty vec if there are none.
+    pub(crate) fn ommers(
+        &self,
+        block_id: impl Into<BlockId>,
+    ) -> EthResult<Option<Vec<reth_primitives::Header>>> {
+        let block_id = block_id.into();
+        Ok(self.client().ommers(block_id)?)
```

Comment:
Starting to think we should have the API operate an LRU cache of all of these historical queries, so we avoid going to the db

Reply:
Should we have something like LRU per table?

Reply:
if we add this to the Database/StateProvider directly, we'd need a sync primitive for the cache which could do more harm than good.

for RPC, I was thinking about adding a Task that has an LRU but communicates via a channel.

So the workflow would be something like:
1. RPC request comes in
2. needs block x
3. sends Command to fetch block x to the cache task
4. cache task looks up LRU or fetches from DB sends response back